splink/analyse_blocking.py

```diff
@@ -139,3 +139,103 @@ def cumulative_comparisons_generated_by_blocking_rules(
     linker._analyse_blocking_mode = False
 
     return br_comparisons
+
+
+def count_comparisons_from_blocking_rule_pre_filter_conditions_sqls(
+    linker: "Linker", blocking_rule: Union[str, "BlockingRule"]
+):
+    if isinstance(blocking_rule, str):
+        blocking_rule = BlockingRule(blocking_rule, sqlglot_dialect=linker._sql_dialect)
+
+    join_conditions = blocking_rule._join_conditions
+
+    l_cols_sel = []
+    r_cols_sel = []
+    l_cols_gb = []
+    r_cols_gb = []
+    using = []
+    for (
+        i,
+        (l_key, r_key),
+    ) in enumerate(join_conditions):
+        l_cols_sel.append(f"{l_key} as key_{i}")
+        r_cols_sel.append(f"{r_key} as key_{i}")
+        l_cols_gb.append(f"{l_key}")
+        r_cols_gb.append(f"{r_key}")
+        using.append(f"key_{i}")
+
+    l_cols_sel = ", ".join(l_cols_sel)
+    r_cols_sel = ", ".join(r_cols_sel)
+    l_cols_gb = ", ".join(l_cols_gb)
+    r_cols_gb = ", ".join(r_cols_gb)
+    using = ", ".join(using)
```

Comment:
Whether you want to is a different matter, but this can be simplified, making this section of the code less verbose and overbearing:
```suggestion
    l_cols_gb, r_cols_gb = zip(*join_conditions)
    using = [f"key_{i}" for i in range(len(join_conditions))]
    l_cols_sel = ", ".join([f"{l_c} as {u}" for l_c, u in zip(l_cols_gb, using)])
    r_cols_sel = ", ".join([f"{r_c} as {u}" for r_c, u in zip(r_cols_gb, using)])
    l_cols_gb = ", ".join(l_cols_gb)
    r_cols_gb = ", ".join(r_cols_gb)
    using = ", ".join(using)
```

Reply:
That's fair! I find it slightly easier to read, but there is no right answer here.

Marking as resolved.